src/rust/rust-proto-new/src/graplinc/common/v1beta1.rs

```diff
@@ -0,0 +1,197 @@
+pub use std::time::{
+    Duration,
+    SystemTime,
+};
+use std::time::{
+    SystemTimeError,
+    UNIX_EPOCH,
+};
+
+use bytes::{
+    Buf,
+    BufMut,
+};
+use prost::Message;
+pub use uuid::Uuid;
+
+use crate::{
+    protobufs::graplinc::common::v1beta1::{
+        Duration as _Duration,
+        Timestamp as _Timestamp,
+        Uuid as _Uuid,
+    },
+    type_url,
+    SerDe,
+    SerDeError,
+};
+
+//
+// Uuid
+//
+
+impl From<_Uuid> for Uuid {
+    fn from(uuid_proto: _Uuid) -> Self {
+        Uuid::from_u128_le(u128::from(uuid_proto.lsb) + (u128::from(uuid_proto.msb) << 64))
+    }
+}
+
+impl From<Uuid> for _Uuid {
+    fn from(uuid: Uuid) -> Self {
+        let bytes = uuid.as_bytes();
+
+        let mut lower: [u8; 8] = Default::default();
+        lower.copy_from_slice(&bytes[..8]);
+
+        let mut upper: [u8; 8] = Default::default();
+        upper.copy_from_slice(&bytes[8..]);
+
+        _Uuid {
+            lsb: u64::from_le_bytes(lower),
+            msb: u64::from_le_bytes(upper),
+        }
+    }
+}
+
+impl type_url::TypeUrl for Uuid {
+    const TYPE_URL: &'static str = "graplsecurity.com/graplinc.common.v1beta1.Uuid";
+}
+
+impl SerDe for Uuid {
```

Comment:
You could just write a blanket impl for T: Message + url::TypeUrl, no need to have this impl duplicated for every type.

Reply:
what is a blanket impl?

Reply:
Oh, sorry, I misunderstood your comment. You were referring to `SerDe` not `TypeUrl`. I'll look into that.

Reply:
I'm not sure this is quite so simple, because some types are `From` and some are `TryFrom`. I've implemented `From` for infallible conversions and `TryFrom` otherwise. I'm not sure how I can accommodate both in one blanket implementation.

Reply:
Actually, thinking about this further.. are you suggesting adding a type parameter to `SerDe`? Like parametrize `SerDe` by the protobuf stub type? This is how the python code works, due to inheritance (I think?) but I was able to get by _without_ doing that here..

Reply:
Hm, yeah nevermind, not worth figuring it out. I think there's a way to perform write it like this is all:
`impl SerDe for T: Message + TypeUrl {`

But it's not important, and I see that your impl uses From/TryFrom, which is probably for the best